.github/workflows/agent-build-refactored.yml

```diff
@@ -163,4 +189,190 @@ jobs:
         with:
           name: ${{ matrix.name }}-registry
           path: /tmp/result
-          retention-days: 1
\ No newline at end of file
+          retention-days: 1
+
+  test-k8s-images:
```

Comment:
I assume old / existing workflow will be removed once that PR is merged and verified?

Reply:
No, it will be removed when it is sure that all existing tests are moved here and also  the new testing framework itself is tested and approved by the rest of the team.